lib/spack/spack/test/architecture.py

```diff
@@ -199,6 +199,9 @@ def test_satisfy_strict_constraint_when_not_concrete(architecture_tuple, constra
 def test_concretize_target_ranges(root_target_range, dep_target_range, result):
     # use foobar=bar to make the problem simpler for the old concretizer
     # the new concretizer should not need that help
+    if spack.config.get("config:concretizer") == "original":
```

Comment:
Any reason not to parameterize (`pytest.mark.parametrize`) this (and tests below) to override the setting and test with `original` and `clingo`?

Reply:
For `pytest.parametrize`, I can look into refactoring this test. but un this PR I just went with the minimal change I had to do to have CI pass.

> test with original and clingo

Maybe I wasn't super clear in the description. This fix actually breaks some behavior for the original concretizer. The behavior that is broken relies on the bug that is fixed here (i.e. if you parse a spec literal with just `target=XX` the spec also contains the platform). If we accept that we can break that behavior in the original concretizer, then the fix is removing code. If we want to fix the original concretizer for the case where it breaks (target ranges in some situations) we need to go back tweaking our greedy algorithm (which might take a while, at least for me).

Reply:
> Maybe I wasn't super clear in the description. This fix actually breaks some behavior for the original concretizer. The behavior that is broken relies on the bug that is fixed here (i.e. if you parse a spec literal with just `target=XX` the spec also contains the platform). If we accept that we can break that behavior in the original concretizer, then the fix is removing code. If we want to fix the original concretizer for the case where it breaks (target ranges in some situations) we need to go back tweaking our greedy algorithm (which might take a while, at least for me).

You were.  I thought it could be useful to have the tests run regardless of the concretizer settings so that both concretizers get exercised by the tests, even with the original getting skipped.

Reply:
> I thought it could be useful to have the tests run regardless of the concretizer settings so that both concretizers get exercised by the tests, even with the original getting skipped.

We run two jobs in CI, one tests with the original concretizer on the other with clingo. I agree it would be better if we can switch seamlessly and have one job - but that requires a big refactoring of code due to our use of globals.